docs/classic-ui/page-templates/index.md

```diff
@@ -0,0 +1,318 @@
+
+
+### Creating your own Templates
+
+If you create your own dexterity object, you probably want to display it on a web browser.  Without a template, a default view will come up using plone's built-in main template and default view for a content type.
+
+
+#### stand-alone templates
+
+```{warning}
+Although templates can be 'stand alone' templates that render a Plone object directly, this is not best practice. A combination of a View and Page Template is the correct implementation.
+```
+
+The simplest possible template is an html document:
+
+template.pt
+```html
+<html>
+<body>
+<h1> hello custom template </h1>
+</body>
+</html>
+```
+
+and registering it in ZCML: 
+
+```xml+genshi
+   <browser:page
+        for="*"
+        name="custom_template"
+        permission="zope.Public"
+        template="template.pt"
+    />
+```
+
+if you now go to any content item on your plone site, including the site itself, and add `custom_template` to the end of the url, the above html will display.
+
+This isn't very useful, but does show how Plone basically works.
+
+#### browser view templates
+
+
+### Advanced Concepts
+
+#### calling templates in python
+
+#### passing data to templates
+
+### Language Reference
+## Chameleon
+
+Chameleon is an HTML/XML template engine for Python.
+
+It’s designed to generate the document output of a web application, typically HTML markup or XML.
+
+The complete documentation for Chameleon can be found at [https://chameleon.readthedocs.io/](https://chameleon.readthedocs.io/) 
+
+The language used is *page templates*, originally a Zope invention[^1].
+
+The template engine compiles templates into Python byte-code and is optimized for speed. For a complex template language, the performance is very good.
+
+The *page templates* language is used within your document structure as special element attributes and text markup. Using a set of simple language constructs, you control the document flow, element repetition, text replacement and translation.
+
+The basic language (known as the *template attribute language* or TAL)
+is simple enough to grasp from an example:
+
+```xml+genshi
+<html>
+  <body>
+    <h1>Hello, ${'world'}!</h1>
+    <table>
+      <tr tal:repeat="row 'apple', 'banana', 'pineapple'">
+        <td tal:repeat="col 'juice', 'muffin', 'pie'">
+           ${row.capitalize()} ${col}
+        </td>
+      </tr>
+    </table>
+  </body>
+</html>
+```
+
+The ``${...}`` notation is shorthand for text insertion. The
+Python-expression inside the braces is evaluated and the result
+included in the output. By default, the string is escaped before
+insertion. To avoid this, use the ``structure:`` prefix:
+
+```xml+genshi
+  <div>${structure: ...}</div>
```

Comment:
This doesn't work as shorthand notation. See https://github.com/zopefoundation/Zope/issues/1077. You have to use `<div tal:content="structure ..." />` instead.

Reply:
Addressed in 56c9b95946ebbf11f93de0423cd2dd1c60d15581